src/vivarium_gates_nutrition_optimization_child/components/risk.py

```diff
@@ -95,3 +98,97 @@ def _get_current_exposure(self, index: pd.Index) -> pd.Series:
             exposure = distribution.ppf(group_df["propensity"])
             exposures.append(exposure)
         return pd.concat(exposures).sort_index()
+
+
+class CGFRiskEffect(RiskEffect):
+    def __init__(self, target: str):
+        """
+        Parameters
+        ----------
+        target :
+            Type, name, and target rate of entity to be affected by risk factor,
+            supplied in the form "entity_type.entity_name.measure"
+            where entity_type should be singular (e.g., cause instead of causes).
+        """
+        self.risk = EntityString("risk_factor.child_growth_failure")
+        self.cgf_models = [
+            EntityString(f"risk_factor.{risk}")
+            for risk in [
+                data_keys.WASTING.name,
+                data_keys.WASTING.name,
+                data_keys.STUNTING.name,
+            ]
+        ]
+        self.target = TargetString(target)
+        self.configuration_defaults = self._get_configuration_defaults()
+
+        self.target_pipeline_name = f"{self.target.name}.{self.target.measure}"
+        self.target_paf_pipeline_name = f"{self.target_pipeline_name}.paf"
+
+    def _get_configuration_defaults(self) -> Dict[str, Dict]:
+        def get_config_by_risk(risk):
+            return {
+                f"effect_of_{risk.name}_on_{self.target.name}": {
+                    self.target.measure: RiskEffect.configuration_defaults[
+                        "effect_of_risk_on_target"
+                    ]["measure"]
+                }
+            }
+
+        config = get_config_by_risk(self.risk)
+        for sub_risk in self.cgf_models:
+            config.update(get_config_by_risk(sub_risk))
+        return config
+
+    def _get_distribution_type(self, builder: Builder) -> str:
+        return "ordered_polytomous"
+
+    def _get_risk_exposure(self, builder: Builder) -> Callable[[pd.Index], pd.Series]:
+        return {
+            risk: builder.value.get_value(f"{risk.name}.exposure") for risk in self.cgf_models
+        }
+
+    def _get_relative_risk_source(self, builder: Builder) -> LookupTable:
+        return {
+            risk: builder.lookup.build_table(
+                get_relative_risk_data(builder, risk, self.target),
+                key_columns=["sex"],
+                parameter_columns=["age", "year"],
+            )
+            for risk in self.cgf_models
+        }
+
+    def _get_population_attributable_fraction_source(self, builder: Builder) -> LookupTable:

```

Comment:
What is the reason you needed to override this method from the superclass?

Reply:
The validation in `get_population_attributable_fraction_data` steps look for a 'configuration.child_growth_failure' key, which we have not defined, since we didn't make ChildGrowthFailure a Risk in its own right.